example/poisson_gmg/poisson_equation.hpp

```diff
@@ -164,9 +184,128 @@ class PoissonEquation {
     return TaskStatus::complete;
   }
 
+  template <class... var_ts>
+  parthenon::TaskID Prolongate(parthenon::TaskList &tl, parthenon::TaskID depends_on,
```

Comment:
Not sure I love moving this into the poisson example. Any way we can keep at least a default prolongation type available publicly?

Reply:
The default Parthenon prolongation is used by default, the user defined prolongation within the equation class only gets called if you set `solver/interior_prolongation=User`.

Reply:
Will the default parthenon prolongation work for multigrid?

Reply:
The default will be the same as it was before this PR, `refinement_ops::ProlongateSharedLinear`, which works for multigrid (although it may not be the optimal prolongation operator). I think what I said before was unclear.

Reply:
The switch between the Parthenon prolongation infrastructure and the user defined prolongation infrastructure is here: https://github.com/parthenon-hpc-lab/parthenon/blob/d554036717275c584991fd33742fca68f37c76b7/src/solvers/mg_solver.hpp#L488. I think it makes more sense to allow for a separate prolongation implementation in this case because it is easier if the prolongation operator is explicitly physical boundary aware.

Reply:
Yeah that makes sense. I assume bounds aware prolongation/restriction ops have to be implemented downstream... doesn't make much sense to implement them upstream.